lib/r10k/environment/name.rb

```diff
@@ -71,8 +71,24 @@ def dirname
 
       private
 
-      def derive_prefix(source,prefix)
+      def derive_name(name, strip_component)
+        if strip_component == nil
```

Comment:
Is `false` a value you could see users put in here? Should we instead handle "falsey" values?

Reply:
Yeah, that would make sense. Updated to handle "falsey" instead of nil.